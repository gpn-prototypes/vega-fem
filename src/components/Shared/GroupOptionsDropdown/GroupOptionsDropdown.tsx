import React from 'react';
import { Button, Dropdown, IconKebab } from '@gpn-prototypes/vega-ui';

import Article from '../../../../types/Article';

import { GroupMenu } from './GroupMenu/GroupMenu';
import { cnGroupOptionsDropdown } from './cn-group-options-dropdown';

import './GroupOptionsDropdown.css';

export interface GroupOptions<GroupType> {
  group: GroupType;
  requestAddArticle: (article: Article, group: GroupType) => void;
  requestChangeGroup?: (group: GroupType) => void;
  requestDeleteGroup: (group: GroupType) => void;
}

export const GroupOptionsDropdown = <GroupType,>({
  group,
  requestAddArticle,
  requestChangeGroup,
  requestDeleteGroup,
}: GroupOptions<GroupType>) => {
  const [isOpen, setIsOpen] = React.useState(false);

  return (
    <Dropdown
      placement="bottom-start"
      isOpen={isOpen}
      onToggle={(nextState): void => {
        setIsOpen(nextState);
      }}
    >
      <Dropdown.Trigger>
        {({ toggle, props: { ref, ...triggerProps } }): React.ReactNode => (
          <Button
            type="button"
            size="s"
            label="Опции"
            onlyIcon
            iconLeft={IconKebab}
            view="ghost"
            ref={ref}
            onClick={toggle}
            {...triggerProps}
          />
        )}
      </Dropdown.Trigger>
      <Dropdown.Menu>
        {({ props: menuProps }): React.ReactNode => (
          <div className={cnGroupOptionsDropdown('menu')} {...menuProps}>
            <GroupMenu
              onClose={() => setIsOpen(false)}
              group={group}
              requestAddArticle={requestAddArticle}
              requestChangeGroup={requestChangeGroup}
              requestDeleteGroup={requestDeleteGroup}
            />
          </div>
        )}
      </Dropdown.Menu>
    </Dropdown>
  );
};
