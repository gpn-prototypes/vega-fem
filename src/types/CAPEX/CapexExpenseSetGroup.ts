import Article, { ArticleValues } from '../Article';

export default interface CapexExpenseSetGroup {
  id?: string | number;
  name?: string;
  caption?: string;
  valueTotal?: number;
  capexExpenseList?: Article[];
  totalValueByYear?: ArticleValues[];
}
