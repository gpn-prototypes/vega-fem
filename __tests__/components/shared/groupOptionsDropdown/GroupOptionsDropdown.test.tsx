import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import { GroupOptionsDropdown } from '../../../../src/components/Shared/GroupOptionsDropdown/GroupOptionsDropdown';
import CapexExpenseSetGroup from '../../../../types/CAPEX/CapexExpenseSetGroup';

let fakeGroup: CapexExpenseSetGroup;

beforeAll(() => {
  fakeGroup = {
    id: '1',
    name: 'oneTimePaymentGroup',
    caption: 'Первоначальный взнос',
  };
});

describe('GroupOptionsDropdown', () => {
  test('рендерится корректно', () => {
    const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
      .create(
        <GroupOptionsDropdown
          group={fakeGroup}
          requestAddArticle={jest.fn()}
          requestChangeGroup={jest.fn()}
          requestDeleteGroup={jest.fn()}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
