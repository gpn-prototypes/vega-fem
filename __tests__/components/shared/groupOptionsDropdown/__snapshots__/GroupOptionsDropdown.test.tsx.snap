// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupOptionsDropdown рендерится корректно 1`] = `
<button
  className="Button Button_size_s Button_view_ghost Button_width_default Button_form_default Button_withIcon Button_onlyIcon MixFocus"
  onClick={[Function]}
  title="Опции"
  type="button"
>
  <span
    className="Icon Icon_size_xs IconKebab Button-Icon"
  >
    <svg
      className="Icon-Svg"
      viewBox="0 0 12 12"
    >
      <path
        d="M7 2a1 1 0 11-2 0 1 1 0 012 0zm0 4a1 1 0 11-2 0 1 1 0 012 0zm-1 5a1 1 0 100-2 1 1 0 000 2z"
      />
    </svg>
  </span>
</button>
`;
